### PR TITLE
DOCTYPE tag is supported by parser

### PR DESCRIPTION
**Pull request by NikolasKomonen**
Refer to #221

***     *****  ***** *****   ****
Some tests need fixing.

doctype formatting tests disabled for this pr. Will be enabled next pr on for doctype formatting.

** **** ** ***    * * *** ***

### CODE REVIEW

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/dom/Node.java [hunk 1]
```diff
@@ -334,11 +334,11 @@ public void setAttributeNode(Attr attr) {
 	}
 
 	/**
-	 * Add node child
+	 * Add node child and set child.parent to this.
 	 * 
 	 * @param child the node child to add.
 	 */
-	public void addChild(Node child) {
+	public void addChildAndSetItsParent(Node child) {
```
Comment (angelozerr) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/dom/Node.java]: Please don't rename this method (keep simple name, and DOM w3c works like this)

**Comment by angelozerr** [comment]
Very good job @NikolasKomonen ! I accept your PR, because I need it and I rename addChid again the method.

It should be really cool if you could implement <!ELEMENT, <!ATTRIBUTE, <!ENTITY and their attributes in order I fix https://github.com/angelozerr/lsp4xml/issues/225

**Comment by NikolasKomonen** [comment]
@angelozerr Great!

> if you could implement <!ELEMENT, <!ATTRIBUTE, <!ENTITY and their attributes

And thats my next task.

**Comment by angelozerr** [comment]
Great! See my comments at https://github.com/angelozerr/lsp4xml/issues/231#issuecomment-439410087

**Comment by angelozerr** [comment]
> And thats my next task.

I'm waiting for that to adjust DTD error ranges, sorry I'm so impatient :)

**Pull request merged**